src/components/heading/heading.test.tsx

```diff
@@ -0,0 +1,239 @@
+  render(
+    <Heading
+      title="foo"
+      help="help text"
+      helpAriaLabel="help-aria-label"
+      helpLink="https://www.warnerbros.com/movies/heat"
+    />
+  );
+
+  const helpIconLink = screen.getByRole("link", { name: "help-aria-label" });
+  expect(helpIconLink).toBeVisible();
+});
+
+test("renders a back link, when `backLink` prop is a href", () => {
+  render(
+    <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+  );
+
+  const backLink = screen.getByRole("link", { name: "Back" });
+  expect(backLink).toHaveAttribute(
+    "href",
+    "https://www.warnerbros.com/movies/heat"
+  );
+});
+
+test("focuses the back link, when `backLink` is a href on mouse down", () => {
+  render(
+    <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+  );
+
+  const backLink = screen.getByRole("link", { name: "Back" });
+  fireEvent.mouseDown(backLink);
+
+  expect(backLink).toBeFocused();
+});
+
+test("renders a back button, when `backLink` prop is a function", () => {
+  const backLinkFunction = jest.fn();
+  render(<Heading title="foo" backLink={backLinkFunction} />);
+
+  const backLink = screen.getByRole("button", { name: "Back" });
+  expect(backLink).toBeVisible();
+});
+
+test("focuses the back link, when `backLink` prop is a function", () => {
+  const backLinkFunction = jest.fn();
+  render(<Heading title="foo" backLink={backLinkFunction} />);
+
+  const backLink = screen.getByRole("button", { name: "Back" });
+  fireEvent.mouseDown(backLink);
+
+  expect(backLink).toBeFocused();
+});
+
+/* styling test for coverage */
+test("alters the grid-column property on the subtitle node, when a back link/button is rendered", () => {
+  render(
+    <Heading
+      title="foo"
+      subheader="bar"
+      backLink="https://www.warnerbros.com/movies/heat"
+    />
+  );
+
+  const subtitle = screen.getByTestId("subtitle");
+
+  expect(subtitle).toHaveStyle("grid-column: 2");
+});
+
+/* styling test for coverage */
+test("alters the outline property on the back link/button, when the back link is focused and consumers opt out of the focus redesign", () => {
+  render(
+    <CarbonProvider focusRedesignOptOut>
+      <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+    </CarbonProvider>
+  );
+
+  const backLink = screen.getByTestId("heading-back-button");
+  backLink.focus();
+
+  expect(backLink).toHaveStyleRule(
+    "outline",
+    "3px solid var(--colorsSemanticFocus500)",
+    { modifier: "button:focus" }
+  );
+});
+
+test("renders a divider when the `divider` prop is true", () => {
+  render(<Heading title="foo" divider />);
+
+  const separator = screen.getByRole("separator");
+  expect(separator).toBeVisible();
+});
+
+test("renders a separator when the `separator` prop is true", () => {
+  render(<Heading title="foo" divider separator />);
+
+  const separators = screen.getAllByRole("separator");
```

Comment:
**comment/suggestion**: I have to admit, I'm not really a fan of this test.

For one thing, all we're saying is that all the `hr` elements (they seem to be the only things to do with the Heading component that will have a role of "separator") that are in the DOM are visible. Literally the only way that could fail is if we had one or more `<hr>` elements in the DOM but hid them with CSS. We could not render any of them and this test will still pass (without actually making a single assertion). [EDIT: actually that's not true, I just looked up the RTL docs and apparently `getAllBy` throws an error if it does't find any, which is what we'd want here. However the other problems I'm going on to describe would remain.] More likely, and just as serious, we could fail to render the separator `<hr>` but render the divider one, and this would still pass (I've just tested this btw - delete the line of the component file that renders the <StyledSeparator>, and all tests still pass).

If we really have to go down this route, at the very least we should assert that the length of `separators` is 2. But that feels too implementation-dependent to me (if we ever get rid of the divider - yes I know that's not likely - we'd have to change this without having actually affected how `separator` behaved, and without any comments it's really not clear what the "magic number" 2 would be doing there).

Since the `StyledSeparator` has a `data-role` I actually think the best thing to do here is just `getByTestId("heading-separator")` and assert that that's visible.